zookeeper-contrib/zookeeper-contrib-zooinspector/src/main/java/org/apache/zookeeper/inspector/manager/ZooInspectorManagerImpl.java

```diff
@@ -713,55 +723,54 @@ public void stop() {
     public List<String> loadNodeViewersFile(File selectedFile)
             throws IOException {
         List<String> result = new ArrayList<String>();
-        if (defaultNodeViewersFile.exists()) {
-            FileReader reader = new FileReader(selectedFile);
-            try {
-                BufferedReader buff = new BufferedReader(reader);
-                try {
-                    while (buff.ready()) {
-                        String line = buff.readLine();
-                        if (line != null && line.length() > 0 && !line.startsWith("#")) {
-                            result.add(line);
-                        }
-                    }
-                } finally {
-                    buff.close();
+        BufferedReader reader = null;
+        try {
+            reader = GetReaderForFile(selectedFile);
+            if(reader == null) {
+                return result;
+            }
+
+            while (reader.ready()) {
```

Comment:
usually you read from the `BufferedReader` until it returns a `null` line

```
String line = reader.readLine();
while (line != null) {
....
line = reader.readLine();

}
```

Reply:
Nice catch.  Looks like that was a subtle bug in the original implementation.  It does do the null check for `readLine`, but it does it inside the loop where it does the `ready()` check.  Seems like the `ready()` check is unnecessary.  I'll remove it.